### PR TITLE
[CELEBORN-1700][FOLLOWUP] Support ShuffleFallbackCount metric for fallback to vanilla Flink built-in shuffle implementation

### PR DESCRIPTION
**Pull request by SteNicholas**
### What changes were proposed in this pull request?

 Support `ShuffleFallbackCount` metric for fallback to vanilla Flink built-in shuffle implementation.

### Why are the changes needed?

#2932 has already supported fallback to vanilla Flink built-in shuffle implementation, which is lack of `ShuffleFallbackCount` metric to feedback the situation of fallback.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

`RemoteShuffleMasterSuiteJ#testRegisterPartitionWithProducerForForceFallbackPolicy`

### CODE REVIEW

**Comment by SteNicholas** [comment]
Ping @reswqa, @codenohup.

**Review by codenohup — APPROVED** [review]

**Comment by codenohup** [comment]
Hi, @SteNicholas 

This PR LGTM, but I have a minor question. It seems that the decision to fall back to the built-in shuffle is based on `job` granularity rather than `Shuffle` granularity. I think it would be beneficial to have a fallback count metric that reflects `job` granularity.

**Comment by SteNicholas** [comment]
@codenohup, The difference of implementation that Flink only supports job granular fallback but Spark supports shuffle granular fallback causes above fallback metric situation. I would like to support job granular fallback metrics in another pull requests

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/fallback/ShuffleFallbackPolicyRunner.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public class ShuffleFallbackPolicyRunner {
   private static final List<ShuffleFallbackPolicy> FALLBACK_POLICIES =
       ShuffleFallbackPolicyFactory.getShuffleFallbackPolicies();
 
-  public static boolean applyFallbackPolicies(
+  public static Optional<ShuffleFallbackPolicy> applyFallbackPolicies(
```
Comment (FMX) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/fallback/ShuffleFallbackPolicyRunner.java:35]: I think this function name can be changed into `getActiveFallbackPolicy` to match its logic.

**Review by FMX — COMMENTED** [review]
LGTM overall except a nit.

**Review by reswqa — APPROVED** [review]
LGTM.

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.6.0).

**Pull request closed without merging**